download_weather_incremental.py

```
@@ -0,0 +1,366 @@
+import time
+import json
+import csv
+import aiohttp
+from threading import Lock
+
+from src.plugins import get_registry
+from src.domain.models import ParameterType, Location
+from src.utils.data_analyzer import analyze_dataset
+
+try:
+    from tqdm.asyncio import tqdm as atqdm
```

Ruff (0.12.2)

`tqdm.asyncio.tqdm` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)